File: app/src/main/java/com/emiryanvl/tapreader/domain/usecases/GetAllBooksUseCase.java
package com.emiryanvl.tapreader.domain.usecases;

import com.emiryanvl.tapreader.domain.model.Book;
import com.emiryanvl.tapreader.domain.repositories.BookRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Observable;

public class GetAllBooksUseCase {

    BookRepository repository;

    @Inject
    public GetAllBooksUseCase(BookRepository repository) {
        this.repository = repository;
    }

    public Observable<List<Book>> execute() {
        return repository.getAllBooks();
    }
}

File: app/src/main/java/com/emiryanvl/tapreader/domain/usecases/GetAllBooksUseCase.kt
package com.emiryanvl.tapreader.domain.usecases

import com.emiryanvl.tapreader.domain.repositories.BookRepository
import javax.inject.Inject

class GetAllBooksUseCase @Inject constructor(private val repository: BookRepository) {
    operator fun invoke() = repository.getAllBooks()
}